New features

sverinn opened the pull request:
Add revive item requirement option
Add downed player looting (RMB)
Add set health after revival
Add carry player with Shift+RMB
Add drop player with Shift+LMB
Minor refactoring

sverinn committed:
Add revive item requirement option
Add downed player looting (RMB)
Add set health after revival
Add carry player with Shift+RMB
Add drop player with Shift+LMB
Minor refactoring

sverinn committed:
Allow multiple valid revive items

sverinn committed:
Fix message spam

sverinn committed:
Fix revive Item check bug

Mynion commented on common/src/main/java/org/mynion/knockoutplugin/listeners/PlayerSneakListener.java:
```diff
@@ -1,38 +1,38 @@
 
-            // Start reviving if present
-            knockedOutPlayer.ifPresent(ko -> NpcManager.startReviving(p, ko));
+        if (reviveItemMaterial != null && reviveItemMaterial.contains(mainHandItem.getType().toString())) {
```
I think it should be 
` if (reviveItemMaterial != null && **!**reviveItemMaterial.contains(mainHandItem.getType().toString())) {`
because otherwise it works opposite as supposed

sverinn replied:
I've reverted revival item check to signle item as a Material, because I couldn't make it consistently check multiple allowed items as String. This might need a helper function, I might finish this in the next feature addition.

Mynion commented on common/src/main/java/org/mynion/knockoutplugin/listeners/EntityInteractListener.java:
```diff
@@ -0,0 +1,75 @@
+        NpcModel npc = npcManager.getNpc((ArmorStand) e.getRightClicked());
+        Player clickedPlayer = npc.getPlayer();
+
+        // Проверяем, разрешен ли лутинг в конфиге
```
Please change all code comments to English only

sverinn replied:
Done

Mynion commented on common/src/main/java/org/mynion/knockoutplugin/listeners/PlayerSneakListener.java:
```diff
@@ -1,38 +1,38 @@
-            // Start reviving if present
-            knockedOutPlayer.ifPresent(ko -> NpcManager.startReviving(p, ko));
+        if (reviveItemMaterial != null && reviveItemMaterial.contains(mainHandItem.getType().toString())) {
+            MessageUtils.sendMessage(player, "revive-item-missing");
```
`if (!player.isSneaking()) MessageUtils.sendMessage(player, "revive-item-missing-message");`
In config the option is named "revive-item-missing-message". Also this "if" statement will prevent sending that message twice.

sverinn replied:
Done

Mynion reviewed (changes requested):
Thanks for contributing, and sorry for the delay in reviewing. I really like the new features! Just a few things to fix before I can merge – I’ve left comments below.

sverinn committed:
Fix revive Item check bug
Change comments to English only
Revert multiple allowed requirement items to one only (I couldn't make it work)
Fix main hand item check

sverinn committed:
Merge branch 'main' into main

Mynion committed:
revived item message fix + revive set health fix

Mynion committed:
revived health change

Mynion committed:
no item message when carrying fix

Mynion committed:
option to disable "carry" and "drop" by clicking

Mynion commented:
I added more options to set health after reviving, an option to disable carrying and dropping by Shift+Click in config and fixed some messages
